threads.js

```diff
@@ -2411,12 +2411,12 @@ Process.prototype.reportLetter = function (idx, string) {
     return str[i - 1] || '';
 };
 
-Process.prototype.reportStringSize = function (string) {
-    if (string instanceof List) { // catch a common user error
-        return string.length();
+Process.prototype.reportStringSize = function (data) {
+    if (data instanceof List) { // catch a common user error
+        return data.length();
     }
-    var str = (string || '').toString();
-    return str.length;
+
+    return (isNil(data) ? '' : data).toString().length;
```

Comment:
It seems to be that these checks are better written explicitly as `isNil`? Is there any reason to prefer the other style other than it being a few less characters?

In this case, `isNil` is the correct choice, but I'm wondering if it would make sense to mostly banish the `||` pattern, since I know we've been bitten by this before.